substrate/utils/frame/benchmarking-cli/src/pallet/writer.rs

```diff
@@ -172,6 +176,120 @@ fn map_results(
 	Ok(all_benchmarks)
 }
 
+// Calculates the total maximum weight of an extrinsic (if present, based on the max. component
+// value) and compares it with the max. extrinsic weight allowed in a single block.
+//
+// `max_extrinsic_weight` & `db_weight` are obtained from the runtime configuration.
+pub(crate) fn sanity_weight_check(
```

Comment:
Conceptually, the way to avoid all of these breaking changes is to do this check in the runtime. Is there a reason why it was chosen in the way that it is now? 

As in, there is a single new runtime API through which you pass in all the computed weights, and it would do any checking necessary for it. 

Ofc this is also a breaking change differently, but possibly an alternative design.

I think the current design is also good, but something to keep in mind.

Reply:
Reading the code, I guess the runtime also doesn't know the maximum value for each component..

Reply:
After looking at it again, the component ranges are coming from the runtime actually: [parameter `list`](https://github.com/paritytech/polkadot-sdk/pull/1493/files#diff-9c1fb75b50895780502e9dc479762b6de55a707e764f1b7e1428eccf2586f31eR719)

I'm investigating another approach